tests/kernel/mem_heap/k_heap_api/src/test_kheap_api.c

```diff
@@ -30,21 +30,35 @@ static void tIsr_kheap_alloc_nowait(void *data)
 
 static void thread_alloc_heap(void *p1, void *p2, void *p3)
 {
+	char *p;
+
 	k_timeout_t timeout = Z_TIMEOUT_MS(200);
 
-	char *p = (char *)k_heap_alloc(&k_heap_test, ALLOC_SIZE_2, timeout);
+	p = (char *)k_heap_alloc(&k_heap_test, ALLOC_SIZE_2, K_NO_WAIT);
+
+	zassert_is_null(p, "k_heap_alloc should fail but not");
+
+	p = (char *)k_heap_alloc(&k_heap_test, ALLOC_SIZE_2, timeout);
+
+	zassert_not_null(p, "k_heap_alloc shoulud not fail");
```

Comment:
Two minor items:
1. Typo "shoulud" -> "should"
2. Optional: I would suggest altering the wording to more clearly indicate what did happen.
    For example: "k_heap_alloc should not fail, but did" or "k_heap_alloc failed as it returned NULL"